server.go

```diff
@@ -673,44 +669,108 @@ func (s *Server) applyCreateShardIfNotExists(m *messaging.Message) (err error) {
 		return ErrRetentionPolicyNotFound
 	}
 
-	// If we can match to an existing shard date range then just ignore request.
-	for _, sh := range rp.Shards {
-		if timeBetweenInclusive(c.Timestamp, sh.StartTime, sh.EndTime) {
-			return nil
-		}
+	// If we can match to an existing shard group date range then just ignore request.
```

Comment:
Yes, you can have two writes come into separate data nodes and want to write data for the same database/policy/time range. If a group doesn't exist yet then they'll both try to create it but we only want the first request to go through. When both requests return then both will have the same group available to write to.